src/nupic/algorithms/Connections.hpp

```diff
@@ -322,7 +324,7 @@ namespace nupic
          *
          * @retval Segment data.
          */
-        SegmentData dataForSegment(const Segment& segment) const;
+        SegmentData& dataForSegment(const Segment& segment);
```

Comment:
Both. Maybe I should split it into two methods: `dataForSegment` and `dataForSegment_`. The public method returns a const reference and the protected method returns a non-const reference.